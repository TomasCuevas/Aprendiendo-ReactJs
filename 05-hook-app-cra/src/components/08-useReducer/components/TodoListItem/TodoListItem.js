import React from 'react'
import PropTypes from 'prop-types';

export const TodoListItem = ({ todo, index, handleRemoveTodo, handleToggleTodo }) => {

    return (
        <div
            className='list__item-container'
            key={todo.id} >
                <li className='list__item' >
                    <p 
                        className={ (todo.done) ? 'list__item-text complete' : 'list__item-text' }
                        onClick={ (e) => handleToggleTodo( todo.id ) } 
                    > 
                            {index + 1}. {todo.description} 
                    </p>
                    <button
                        className='list__delete-item'
                        onClick={ () => handleRemoveTodo( todo.id ) } 
                    >
                            Borrar
                    </button>
                </li>
        </div>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
    handleRemoveTodo: PropTypes.func.isRequired,
    handleToggleTodo: PropTypes.func.isRequired
}
