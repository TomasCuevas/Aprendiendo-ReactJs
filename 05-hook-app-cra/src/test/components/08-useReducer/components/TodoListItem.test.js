import { shallow } from "enzyme";

import { TodoListItem } from "../../../../components/08-useReducer/components/TodoListItem/TodoListItem";
import { demoTodos } from '../../../fixtures/demoTodos';

describe('Pruebas en <TodoListItem />', () => {

  const handleRemoveTodo = jest.fn();
  const handleToggleTodo = jest.fn();
  
  const wrapper = shallow(
    <TodoListItem 
      todo={ demoTodos[0] } 
      index={ demoTodos[0].id } 
      handleRemoveTodo={ handleRemoveTodo } 
      handleToggleTodo={ handleToggleTodo }
    />
  ) 

  test('debe de mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot();  
  });

  test('debe de llamar la funcion handleRemove', () => {
    wrapper.find('button').simulate('click');
    expect(handleRemoveTodo).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('debe de llamar la funcion handleToggle', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggleTodo).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test('debe de mostrar el texto correctamente', () => {
    expect( wrapper.find('p').text() ).toBe(`${demoTodos[0].id + 1}. ${demoTodos[0].description}`);
  });

  test('debe de tener la clase "complete" si TODO.ID = true', () => {
    const todo = demoTodos[0];
    todo.done = true;
    
    const wrapper = shallow(
      <TodoListItem 
        todo={ todo } 
        index={ todo.id } 
        handleRemoveTodo={ handleRemoveTodo } 
        handleToggleTodo={ handleToggleTodo }
      />
    ) 
    
    expect( wrapper.find('p').hasClass('complete') ).toBe(true);
  });
})