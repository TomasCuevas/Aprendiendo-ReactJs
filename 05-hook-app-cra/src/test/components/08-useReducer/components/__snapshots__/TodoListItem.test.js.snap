// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> debe de mostrarse correctamente 1`] = `
<div
  className="list__item-container"
  key="1"
>
  <li
    className="list__item"
  >
    <p
      className="list__item-text"
      onClick={[Function]}
    >
      2
      . 
      Aprender React
    </p>
    <button
      className="list__delete-item"
      onClick={[Function]}
    >
      Borrar
    </button>
  </li>
</div>
`;
